module/mempool/stdmap/backend.go

```diff
@@ -100,16 +100,18 @@ func (b *Backdata) Hash() flow.Identifier {
 type Backend struct {
 	sync.RWMutex
 	Backdata
-	limit uint
-	eject EjectFunc
+	limit             uint
+	eject             EjectFunc
+	ejectionCallbacks []OnEjection
```

Comment:
```suggestion
	ejectionCallback OnEjection
```

I think we could just have one ejectionCallback. So the caller have to provide the ejection callback as constructor argument, if not, the default ejection callback will just be a noop. This eliminates the case where the ejectionCallback is `nil`.

But what if there are multiple modules are interested when an entity is ejected and wanted to receive ejection callback?

We could call into multiple callbacks in the one ejection callback:

```
callback := func(entity flow.Entity) {
  a.OnEntityEjected(entity)
  b.OnEntityEjected(entity)
  c.OnEntityEjected(entity)
}
backend := NewBackend(callback)
```

There are 2 benefits:
1. simple 
2. flexible
The caller can even call multiple callbacks concurrently.

Reply:
🎵 thanks for the great suggestion. Implemented